packages/backend/src/core/tx-count/RpcBlockDownloader.ts

```diff
@@ -0,0 +1,67 @@
+import { Logger, TaskQueue } from '@l2beat/common'
+import { ProjectId, UnixTime } from '@l2beat/types'
+
+import { TxCountRepository } from '../../peripherals/database/TxCountRepository'
+import { EthereumClient } from '../../peripherals/ethereum/EthereumClient'
+import { Clock } from '../Clock'
+
+export class RpcBlockDownloader {
+  private taskQueue = new TaskQueue<void>(() => this.update(), this.logger)
+  private blockQueue = new TaskQueue(this.getBlock.bind(this), this.logger, 100)
+
+  constructor(
+    private etheteumClient: EthereumClient,
+    private txCountRepository: TxCountRepository,
+    private clock: Clock,
+    private logger: Logger,
+    private projectId: ProjectId,
+  ) {
+    this.logger = logger.for(this)
+  }
+
+  async getBlock(number: number) {
+    const block = await this.etheteumClient.getBlock(number)
+    const timestamp = new UnixTime(block.timestamp)
+
+    if (timestamp.gt(this.clock.getLastHour())) {
```

Comment:
I do not quite get this check, how are we able to fetch block greater than .getLastHour() ? Or is it just sanity check?

Reply:
This is basically why this feature is tricky. We do not hold timestamp -> blockNumber mapping in the database for all the networks. Current idea is to download all the blocks (until eth_blockNumber) and then discard all that happened after clock.getLastHour(), to avoid dealing with reorgs.

Reply:
This deserves in code comment